src/doc/src/reference/build-script-examples.md

```diff
@@ -0,0 +1,501 @@
+## Build Script Examples
+
+The following sections illustrate some examples of writing build scripts.
+
+Some common build script functionality can be found via crates on [crates.io].
+Check out the [`build-dependencies`
+keyword](https://crates.io/keywords/build-dependencies) to see what is
+available. Some popular crates are:
+
+* [`bindgen`](https://crates.io/crates/bindgen) — Automatically generate Rust
+  FFI bindings to C libraries.
+* [`cc`](https://crates.io/crates/cc) — Compiles C/C++/assembly.
+* [`pkg-config`](https://crates.io/crates/pkg-config) — Detect system
+  libraries using the `pkg-config` utility.
+* [`cmake`](https://crates.io/crates/cmake) — Runs the `cmake` build tool to build a native library.
+* [`rustc_version`](https://crates.io/crates/rustc_version),
+  [`version_check`](https://crates.io/crates/version_check) — These crates
+  provide ways to implement conditional compilation based on the current
+  version of `rustc`.
```

Comment:
I'm slightly hesitant to recommend these because there's actually quite a few on crates.io, and some (like `rustc_version`) have known drawbacks and hindrances in terms of compile time and/or robustness of the parser. I'd also personally recommend `autocfg`, but maybe we should just maintain a list of all of them here?

Reply:
Yea, it's tough because I don't want it to seem too much like an endorsement or recommendation, but more like "here are some popular examples".  What if there was a disclaimer of some kind?  I included those two version crates just because they have such a huge number of downloads, and I figure the reader can decide which one is right for them.  

I think I just forgot to add `autocfg`.

I think a warning about compile times is a good idea.

Reply:
Ok, sounds reasonable to me!

Reply:
Updated.